### DISCUSSION TITLE
Inefficient first-time fetching of values for bulk subscriptions

### DISCUSSION THREAD

**Discussion by MrCaoBing** [eclipse-milo/milo#1040] [Ideas]
Hi, I have a problem that when I start a bulk subscription, it takes too long to refresh all nodes, because the current logic of bulk subscription is that the values of all bulk subscribed nodes are read first, regardless of whether the node values have changed or not.

_**org.eclipse.milo.opcua.sdk.client.subscriptions.createMonitoredItems(TimestampsToReturn timestampsToReturn,
        List<MonitoredItemCreateRequest> itemsToCreate,
        ItemCreationCallback itemCreationCallback);**_

The above function callbacks are not asynchronous, is it feasible to change the subscription callbacks to asynchronous processing?

**Comment by kevinherron** [comment]
Can you better explain what you're doing? This callback exists just for attaching a listener to the item, and there's no need to manually refresh anything when you create a monitored item. The server will send an initial value.

**Reply by MrCaoBing** [reply]
When creating multiple monitored items, for example 2000, it will take a long time to process.

**Reply by kevinherron** [reply]
What takes a long time? Have you looked at a Wireshark capture? If the server takes a long time to create 2000 items there is nothing you can do about that from the client.
